pkg/cmd/target/ssh/create/create.go

```diff
@@ -0,0 +1,399 @@
+		},
+	}
+}
+
+func NewCmdCreate(f factory.Factory) *cobra.Command {
+	createFlags := NewCreateFlags()
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a ssh deployment target",
+		Long:  "Create a ssh deployment target in Octopus Deploy",
+		Example: fmt.Sprintf(heredoc.Doc(`
+			$ %s deployment-target ssh create
+		`), constants.ExecutableName),
+		RunE: func(c *cobra.Command, _ []string) error {
+			opts := NewCreateOptions(createFlags, cmd.NewDependencies(f, c))
+
+			return createRun(opts)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&createFlags.Name.Value, createFlags.Name.Name, "n", "", "A short, memorable, unique name for this Deployment Target.")
+	flags.StringVar(&createFlags.Account.Value, createFlags.Account.Name, "", "The name or id of the SSH Keypair or Username/Password account")
+	flags.StringVar(&createFlags.HostName.Value, createFlags.HostName.Name, "", "The hostname or IP address of the deployment target to connect to.")
+	flags.StringVar(&createFlags.Fingerprint.Value, createFlags.Fingerprint.Name, "", "The host fingerprint of the deployment target.")
+	flags.IntVar(&createFlags.Port.Value, createFlags.Port.Name, 0, "The port to connect to the deployment target on.")
+	flags.StringVar(&createFlags.Runtime.Value, createFlags.Runtime.Name, "", fmt.Sprintf("The runtime to use to run Calamari on the deployment target. Options are '%s' or '%s'", SelfContainedCalamari, MonoCalamari))
+	flags.StringVar(&createFlags.Platform.Value, createFlags.Platform.Name, "", fmt.Sprintf("The platform to use for the %s Calamari. Options are '%s', '%s', '%s' or '%s'", SelfContainedCalamari, LinuxX64, LinuxArm64, LinuxArm, OsxX64))
+
+	shared.RegisterCreateTargetEnvironmentFlags(cmd, createFlags.CreateTargetEnvironmentFlags)
+	shared.RegisterCreateTargetRoleFlags(cmd, createFlags.CreateTargetRoleFlags)
+	shared.RegisterCreateTargetProxyFlags(cmd, createFlags.CreateTargetProxyFlags)
+	shared.RegisterCreateTargetMachinePolicyFlags(cmd, createFlags.CreateTargetMachinePolicyFlags)
+	shared.RegisterCreateTargetTenantFlags(cmd, createFlags.CreateTargetTenantFlags)
+	shared.RegisterWebFlag(cmd, createFlags.WebFlags)
+
+	return cmd
+}
+
+func createRun(opts *CreateOptions) error {
+	if !opts.NoPrompt {
+		if err := PromptMissing(opts); err != nil {
+			return err
+		}
+	}
+
+	envs, err := executionscommon.FindEnvironments(opts.Client, opts.Environments.Value)
+	if err != nil {
+		return err
+	}
+	environmentIds := util.SliceTransform(envs, func(e *environments.Environment) string { return e.ID })
+
+	account, err := getAccount(opts)
+	if err != nil {
+		return err
+	}
+
+	port := opts.Port.Value
+	if port == 0 {
+		port = 22
+	}
+	endpoint := machines.NewSSHEndpoint(opts.HostName.Value, port, opts.Fingerprint.Value)
+	endpoint.AccountID = account.GetID()
+
+	if opts.Runtime.Value == SelfContainedCalamari {
+		endpoint.DotNetCorePlatform = opts.Platform.Value
+	} else {
+		endpoint.DotNetCorePlatform = ""
```

Comment:
Is this necessary?

Reply:
yes, if the user provides the mono runtime option, plus the platform

Reply:
I meant, isn't the default value an empty string?

Reply:
Yes, but if both values are provided, should we take the runtime value as the source of truth, or is that a validation error?

Reply:
correct, this is redundant